File: backend/src/routes/v1/pdf.js
'use strict';

const express = require('express');
const router  = express.Router();
const vision  = require('../../services/visionClient');

/**
 * POST /api/v1/pdf/diet
 * Body: { dietData, userName }
 * Proxy al servicio Python → devuelve el PDF al cliente.
 */
router.post('/diet', async (req, res) => {
  const { dietData, userName = 'Usuario' } = req.body;
  if (!dietData) return res.status(400).json({ error: 'dietData es requerido' });

  const pdfBuffer = await vision.generateDietPdf(dietData, userName);
  if (!pdfBuffer) {
    return res.status(503).json({ error: 'Servicio PDF no disponible' });
  }

  const week = dietData.weekStart || dietData.week_start || 'semana';
  res.set({
    'Content-Type':        'application/pdf',
    'Content-Disposition': `attachment; filename="fittracker-dieta-${week}.pdf"`,
    'Content-Length':      pdfBuffer.length,
  });
  res.send(pdfBuffer);
});

module.exports = router;

File: python_service/app/routers/pdf.py
"""
PDF Router — Genera y descarga el plan de dieta semanal en PDF.
POST /pdf/diet   { diet_data: {...}, user_name: "..." }
"""
from fastapi import APIRouter, HTTPException
from fastapi.responses import Response
from pydantic import BaseModel
from typing import Any

from app.auth import verify_internal_token
from fastapi import Depends

router = APIRouter(prefix="/pdf", tags=["pdf"])


class DietPdfRequest(BaseModel):
    diet_data: dict[str, Any]
    user_name: str = "Usuario"


@router.post(
    "/diet",
    dependencies=[Depends(verify_internal_token)],
    response_class=Response,
    responses={200: {"content": {"application/pdf": {}}}},
)
async def generate_diet_pdf(payload: DietPdfRequest):
    try:
        from app.services.pdf_service import generate_diet_pdf as _gen
        pdf_bytes = _gen(payload.diet_data, payload.user_name)
        week = payload.diet_data.get("weekStart") or payload.diet_data.get("week_start", "semana")
        filename = f"fittracker-dieta-{week}.pdf"
        return Response(
            content=pdf_bytes,
            media_type="application/pdf",
            headers={"Content-Disposition": f'attachment; filename="{filename}"'},
        )
    except Exception as e:
        raise HTTPException(status_code=500, detail=f"Error generando PDF: {e}")
